src/Producer/SyncProcess.php

```diff
@@ -48,11 +48,13 @@ public function send(array $data): array
         //  partId:
         //  key:
         //  value:
-        if (empty($data)) {
+        if (empty($messagesData)) {
             return [];
         }
 
-        $sendData = $this->convertMessage($data);
+        $this->validateMessagesData($messagesData);
```

Comment:
With this we'll be iterating the record set twice (when all records are valid), which is not really optimal...

Reply:
Why did I do that 🤔